jormungandr/cpp/autodiff/BindVariableMatrices.cpp

```diff
@@ -147,11 +147,23 @@ void BindVariableMatrix(py::module_& autodiff,
   variable_matrix.def(
       "__getitem__",
       [](VariableMatrix& self,
-         py::tuple slices) -> VariableBlock<VariableMatrix> {
+         py::tuple slices) -> py::object {  // Change return type to py::object
         if (slices.size() != 2) {
           throw py::index_error(
               fmt::format("Expected 2 slices, got {}.", slices.size()));
         }
+        // Check if both indices are integers (not slices)
+        if (py::isinstance<py::int_>(slices[0]) &&
+            py::isinstance<py::int_>(slices[1])) {
+          int row = slices[0].cast<int>();
+          int col = slices[1].cast<int>();
+          // Bounds checking
+          if (row >= self.Rows() || col >= self.Cols()) {
+            throw std::out_of_range("Index out of bounds");
+          }
+          // Return single Variable
+          return py::cast(self(row, col));
+        }
```

Comment:
```suggestion
        // If both indices are integers instead of slices, return Variable instead of VariableBlock
        if (py::isinstance<py::int_>(slices[0]) &&
            py::isinstance<py::int_>(slices[1])) {
          int row = slices[0].cast<int>();
          int col = slices[1].cast<int>();

          if (row >= self.Rows() || col >= self.Cols()) {
            throw std::out_of_range("Index out of bounds");
          }

          return py::cast(self(row, col));
        }
```